nbgrader/preprocessors/getgrades.py

```diff
@@ -1,9 +1,10 @@
 from .. import utils
 from ..api import Gradebook
+from ..plugins import LateSubmissionPlugin
 from . import NbGraderPreprocessor
 
 
-class GetGrades(NbGraderPreprocessor):
+class GetGrades(NbGraderPreprocessor, LateSubmissionPlugin):
```

Comment:
Rather than inheriting from the plugin here, I would add a new preprocessor whose job it is to handle late penalties -- maybe called `AssignLatePenalties` -- and then add that preprocessor to the autograde app. That way late penalties are assigned during autograding, rather than during feedback (since not everyone might use nbgrader-generated feedback, but might want to rely on nbgrader to calculate late penalties).